### PR TITLE
Added check for `window` vs `global` on npm node runtime package, per #53

### PR DESCRIPTION
**Pull request by niftylettuce**

### CODE REVIEW

**Comment by niftylettuce** [comment]
OK it's ready.

File: fetch-npm-node.js [hunk 1]
```diff
@@ -1,16 +1,21 @@
 "use strict";
 
-var realFetch = require('node-fetch');
-module.exports = function(url, options) {
-	if (/^\/\//.test(url)) {
```
Comment (matthew-andrews) [inline fetch-npm-node.js]: Maybe you disagree with why this was added — so that the client side and server side interface was the same (you can make requests to `//domain.com` and both would work) but I don't think we need to remove this at this stage — would require a major version bump.

File: fetch-npm-node.js [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
-if (!global.fetch) {
-	global.fetch = module.exports;
-	global.Response = realFetch.Response;
-	global.Headers = realFetch.Headers;
-	global.Request = realFetch.Request;
+// add support for jsdomify for running tests
+// <https://github.com/niftylettuce/frisbee>
+// <https://github.com/podio/jsdomify>
+// <https://github.com/github/fetch/issues/238>
+if (typeof window === 'object') {
```
Comment (matthew-andrews) [inline fetch-npm-node.js]: I don't understand … this file should never be required in a client side context … that's why the `browser` property is set … what am I missing?

Comment (niftylettuce) [inline fetch-npm-node.js]: check out my browser test on my package frisbee to see why - it doesn't
take into consideration for tests

File: package.json [hunk 3]
```diff
@@ -20,13 +20,13 @@
   "homepage": "https://github.com/matthew-andrews/isomorphic-fetch/issues",
   "dependencies": {
     "node-fetch": "^1.0.1",
-    "whatwg-fetch": ">=0.10.0"
+    "whatwg-fetch": "niftylettuce/fetch"
```
Comment (matthew-andrews) [inline package.json]: for various reasons I'd prefer to keep this on the main version — not on someone's fork.  Why is this needed?

File: fetch-npm-node.js (hunk 2 above)
Comment (matthew-andrews) [inline fetch-npm-node.js]: Hmm still doesn't add up. If you're running tests in a browser environment should still hit the browser file

Comment (matthew-andrews) [inline fetch-npm-node.js]: If not that's a big in your test library

Comment (matthew-andrews) [inline fetch-npm-node.js]: bug*

File: package.json (hunk 3 above)
Comment (niftylettuce) [inline package.json]: because they use `self` instead of `window` and tests don't pass when you
try to simulate browser with jsdom/jsdomify.  though I could be wrong - I
have to rewrite with pure `jsdom` instead of using `jsdomify` per
https://github.com/tmpvar/jsdom/issues/1203#issuecomment-160725194

Comment (niftylettuce) [inline package.json]: don't accept this PR till I look into that jsdom#1203 more, thanks @matthew-andrews

Comment (matthew-andrews) [inline package.json]: Really sounds like a bug in the browser simulation — `self` should be defined.

**Comment by JohnBaileyN** [comment]
:+1:

**Comment by sofuckingnice** [comment]
up

**Comment by matthew-andrews** [comment]
I believe @niftylettuce was looking for a fix in other libraries.  This PR probably won't get merged in its current state.
